### ISSUE TITLE
make markdown indicator case-insensitive

### ISSUE THREAD

**Issue by kareila** [dreamwidth/dw-free#2419]
Reported on Discord:

> My first two attempts at using Markdown failed because I used a capital M without having the reference right in front of my eyes (silly me, I thought it would be easy to remember...) so would it be reasonable to allow the variant !Markdown to do the same thing as !markdown?

Pretty sure this is just a regular expression that needs the case-insensitive modifier added.

**Comment by alexwlchan** [comment]
A quick grep through the codebase suggests there are three places where this can plausibly occur, which look slightly different to each other:

```console
$ grep -rnI '!markdown' *
cgi-bin/LJ/CleanHTML.pm:1486:    if ($$ref =~ s/^\s*!markdown\s*\r?\n//s) {
cgi-bin/DW/External/XPostProtocol/LJXMLRPC.pm:348:    if ( $entry->event_raw =~ /^\s*!markdown\s*\r?\n/s ) {
cgi-bin/DW/External/XPostProtocol/LJXMLRPC.pm:458:    if ( $event_text =~ s/^\s*!markdown\s*\r?\n//s ) {
```

It would be nice to have a function somewhere `is_this_markdown()` that defines this regex once and only once.

**Comment by kareila** [comment]
Our Markdown support is very slapdash so lots of room for improvement here. But 2/3 of these expressions don’t just pattern match, they also remove the pattern from the displayed text.
